### PR TITLE
Smarten BLAS_VENDOR detection in linear_algebra benchmarks

### PR DESCRIPTION
**Pull request by steven-johnson**
we previously failed if one-but-not-all entries in BLAS_VENDORS was found.

### CODE REVIEW

File: apps/linear_algebra/benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -54,24 +54,31 @@ if (NOT CBLAS_FOUND)
...
-       ${CBLAS_INCLUDE_DIR}
-    )
-    string(TOUPPER ${BLAS_VENDOR} DEFINE_SUFFIX)
-    target_compile_definitions(${TARGET} PRIVATE -DUSE_${DEFINE_SUFFIX} ${MISC_DEFINITIONS})
-    target_compile_options(${TARGET} PRIVATE -Wno-error=unused-variable)
-    target_link_libraries(${TARGET}
-      PRIVATE
-       ${BLAS_${NAME}_LIBRARY}
-       ${${BLAS_VENDOR}_EXTRA_LIBS}
+    set(BLA_VENDOR ${BLAS_VENDOR})
```
Comment (zvookin) [inline apps/linear_algebra/benchmarks/CMakeLists.txt]: BLA_VENDOR?

Comment (steven-johnson) [reply apps/linear_algebra/benchmarks/CMakeLists.txt]: Same pattern that is used in apps/linear_algebra/CMakeLists.txt

Comment (steven-johnson) [reply apps/linear_algebra/benchmarks/CMakeLists.txt]: (May actually be unused. I can check.)

Comment (steven-johnson) [reply apps/linear_algebra/benchmarks/CMakeLists.txt]: Yeah, something is still weird. Investigating.

Comment (steven-johnson) [reply apps/linear_algebra/benchmarks/CMakeLists.txt]: BLA_VENDOR is a thing; see https://cmake.org/cmake/help/v3.0/module/FindBLAS.html

**Pull request merged**